### PR TITLE
Various small improvements

### PR DESCRIPTION
**Pull request by cantino**
* There is a button to add a new Agent directly in a Scenario
* Deleting an Agent from its show page no longer 404s
* There is a button to destroy all delayed_jobs

Code review appreciated @dsander or @knu!

### CODE REVIEW

File: app/controllers/agents_controller.rb [hunk 1]
```diff
@@ -237,14 +240,14 @@ def redirect_back(message, options = {})
       if @agent && !@agent.destroyed?
         path = agent_path(@agent)
```
Comment (knu) [inline app/controllers/agents_controller.rb]: Could you please add `else path = agents_path` here?  It should fix the 404 bug.

File: app/controllers/agents_controller.rb [hunk 2]
```diff
@@ -237,14 +240,14 @@ def redirect_back(message, options = {})
...
       end
-    when /\A#{Regexp::escape scenarios_path}\/\d+\Z/, agents_path
+    when /\A#{Regexp::escape scenarios_path}\/\d+\z/, agents_path
       path = ret
     end
 
     if path
       redirect_to path, notice: message
     else
-      super agents_path, notice: message
```
Comment (knu) [inline app/controllers/agents_controller.rb]: I think this `super` was necessary to get you back to the page you came from when you hit a button on the second page of your agents list, for example. (Sorry if I didn't add the spec for that)

**Comment by knu** [comment]
@cantino Looks great, but I think the 404 bug was introduced by me and I know how to fix it.  If my line notes above do not apply, you can just merge this and then I'll follow up with an alternative fix later.

**Comment by cantino** [comment]
Thanks @knu, I made your changes and tested that they work.  Everything else look good?

**Comment by knu** [comment]
@cantino Yes!

**Pull request merged**